ext/session/session.c

```diff
@@ -369,20 +373,19 @@ PHPAPI zend_result php_session_valid_key(const char *key) /* {{{ */
 				|| (c >= '0' && c <= '9')
 				|| c == ','
 				|| c == '-')) {
```

Comment:
Another way to check for the nul byte is to have the check here, and instead of looping for the first nul byte have a for that uses the zend_string length.

Reply:
I chose this solution. It's the fastest one.